src/Form/matching.js

```diff
@@ -500,24 +503,83 @@ class Matching {
     }
 }
 
+/**
+ *  @returns {SupportedTypes}
+ */
+function getInputType (input) {
+    return input.getAttribute(ATTR_INPUT_TYPE) || 'unknown'
+}
+
+/**
+ * Retrieves the main type
+ * @param {SupportedSubTypes | string} type
+ * @returns {SupportedMainTypes}
+ */
+function getMainTypeFromType (type) {
+    const mainType = type.split('.')[0]
+    switch (mainType) {
+    case 'credentials':
+    case 'creditCard':
+    case 'identities':
+        return mainType
+    }
+    return 'unknown'
+}
+
 /**
  * Retrieves the input main type
  * @param {HTMLInputElement} input
- * @returns {SupportedMainTypes | string}
+ * @returns {SupportedMainTypes}
  */
 const getInputMainType = (input) =>
-    input.getAttribute(ATTR_INPUT_TYPE)?.split('.')[0] ||
-    'unknown'
+    getMainTypeFromType(getInputType(input))
+
+/**
+ * Retrieves the subtype
+ * @param {SupportedTypes | string} type
+ * @returns {SupportedSubTypes | 'unknown'}
+ */
+function getSubtypeFromType (type) {
+    const mainType = type?.split('.')[0]
+    switch (mainType) {
+    case 'emailAddress':
+    case 'password':
+    case 'username':
+    case 'cardName':
+    case 'cardNumber':
+    case 'cardSecurityCode':
+    case 'expirationMonth':
+    case 'expirationYear':
+    case 'expiration':
+    case 'firstName':
+    case 'middleName':
+    case 'lastName':
+    case 'fullName':
+    case 'phone':
+    case 'addressStreet':
+    case 'addressStreet2':
+    case 'addressCity':
+    case 'addressProvince':
+    case 'addressPostalCode':
+    case 'addressCountryCode':
+    case 'birthdayDay':
+    case 'birthdayMonth':
+    case 'birthdayYear':
+        return mainType
+    }
+    return 'unknown'
+}
 
 /**
  * Retrieves the input subtype
  * @param {HTMLInputElement|Element} input
- * @returns {SupportedSubTypes | string}
+ * @returns {SupportedSubTypes | SupportedMainTypes}
  */
-const getInputSubtype = (input) =>
-    input.getAttribute(ATTR_INPUT_TYPE)?.split('.')[1] ||
-    input.getAttribute(ATTR_INPUT_TYPE)?.split('.')[0] ||
```

Comment:
The fallback to the main type seems fragile and doesn't appear to be used. Perhaps there's a use-case for it?